src/main/scala/com/fortysevendeg/android/sqlite/statement/SQLDroidStatement.scala

```diff
@@ -0,0 +1,196 @@
+package com.fortysevendeg.android.sqlite.statement
+
+import java.sql._
+
+import com.fortysevendeg.android.sqlite.logging.{AndroidLogWrapper, LogWrapper}
+import com.fortysevendeg.android.sqlite.resultset.SQLDroidResultSet
+import com.fortysevendeg.android.sqlite.{SQLDroidConnection, SQLDroidDatabase, WrapperNotSupported, _}
+
+import scala.collection.mutable
+
+class SQLDroidStatement(
+  sqlDroidConnection: SQLDroidConnection,
+  columnGenerated: Option[String] = None,
+  logWrapper: LogWrapper = new AndroidLogWrapper()) extends Statement with WrapperNotSupported {
+
+  val selectRegex = "(?m)(?s)\\s*(SELECT|PRAGMA|EXPLAIN QUERY PLAN).*".r
+
+  val connectionClosedErrorMessage = "Connection is closed"
+
+  val statementClosedErrorMessage = "Statement is closed"
+
+  val maxNegativeErrorMessage = "Max rows must be zero or positive"
+
+  val selectLastRowId = "SELECT last_insert_rowid()"
+
+  private[this] val batchList = new mutable.MutableList[String]
+
+  private[this] var connection: Option[SQLDroidConnection] = Option(sqlDroidConnection)
+
+  private[this] var maxRows: Option[Int] = None
+
+  private[this] var resultSet: Option[ResultSet] = None
+
+  private[this] var updateCount: Option[Int] = None
+
+  def getBatchList: Seq[String] = batchList.toList
+
+  override def addBatch(sql: String): Unit = batchList += sql
+
+  override def clearBatch(): Unit = batchList.clear()
+
+  override def close(): Unit = {
+    connection = None
+    closeResultSet()
+  }
+
+  override def isClosed: Boolean = connection.isEmpty
+
+  override def execute(sql: String): Boolean = withOpenConnection { db =>
+    resultSet = selectRegex.pattern.matcher(sql).matches() match {
+      case true =>
+        updateCount = None
+        val limitedSql = maxRows map (m => s"$sql LIMIT $m") getOrElse sql
+        Some(new SQLDroidResultSet(db.rawQuery(limitedSql)))
+      case false =>
+        db.execSQL(sql)
+        updateCount = Option(db.changedRowCount())
+        None
+    }
+    resultSet.isDefined
+  }
+
+  override def executeBatch(): scala.Array[Int] = withOpenConnection { db =>
+    val updateArray = batchList map { sql =>
+      db.execSQL(sql)
+      db.changedRowCount()
+    }
+    updateCount = updateArray.reduceOption { (elem1, elem2) =>
+      (elem1, elem2) match {
+        case (a, b) if a > 0 && b > 0 => a + b
+        case (a, b) if a > 0 => a
+        case (a, b) if b > 0 => b
+        case _ => 0
+      }
+    }
+    updateArray.toArray
+  }
+
+  override def executeQuery(sql: String): ResultSet =
+    withOpenConnection(newQueryResultSet(_, sql))
+
+  override def executeUpdate(sql: String): Int = withOpenConnection { db =>
+    db.execSQL(sql)
+    val count = db.changedRowCount()
+    updateCount = Some(count)
+    count
+  }
+
+  override def getConnection: Connection =
+    connection match {
+      case Some(c) => c
```

Comment:
On top of the previous comment, can exist a kind of danger if we provide to the client a closed connection?